pkg/sdk/functions_ext.go

```diff
@@ -69,9 +92,135 @@ func functionDetailsFromRows(rows []FunctionDetail) (*FunctionDetails, error) {
 			v.TargetPath = row.Value
 		}
 	}
+	if e := errors.Join(errs...); e != nil {
+		return nil, e
+	}
+
+	if functionDetailsImports, err := parseFunctionDetailsImport(*v); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.NormalizedImports = functionDetailsImports
+	}
+
+	if v.TargetPath != nil {
+		if p, err := parseStageLocationPath(*v.TargetPath); err != nil {
+			errs = append(errs, err)
+		} else {
+			v.NormalizedTargetPath = p
+		}
+	}
+
+	if dt, returnNotNull, err := parseFunctionOrProcedureReturns(v.Returns); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.ReturnDataType = dt
+		v.ReturnNotNull = returnNotNull
+	}
+
+	if args, err := parseFunctionOrProcedureSignature(v.Signature); err != nil {
+		errs = append(errs, err)
+	} else {
+		v.NormalizedArguments = args
+	}
+
 	return v, errors.Join(errs...)
 }
 
+func parseFunctionDetailsImport(details FunctionDetails) ([]NormalizedPath, error) {
```

Comment:
nit: I believe this could be simplified with `sdk.ParseCommaSeparatedStringArray` + `collections.MapErr` combo

Reply:
also `parseFunctionOrProcedureSignature` (but with different cutset), but let's leave those as they are

Reply:
They both could, I will leave a TODO about trying to merge with the existing helpers - I was anticipating here a bit different logic, that's why initially I have used a new one.